documentation/src/main/asciidoc/topics/proc_using_multiple_storage_formats.adoc

```diff
@@ -0,0 +1,29 @@
+[id='mediatype_override-{context}']
+= Using Multiple Storage Formats for Custom Code
+You can deploy and run custom code on {brandname}, such as tasks, listeners,
+and merge policies. Custom code on {brandname} can directly access data but 
+must also interoperate with clients that access the same data through different
+endpoints. For example, you can create tasks that handle custom objects while
+Hot Rod clients read and write data in binary format.
+
+In this case, you can configure `application/x-protostream` as the cache
+encoding to store data in binary format then configure your custom code to
+perform cache operations using a different MediaType.
+
+For example:
+
+[source,java]
+----
+include::code_examples/EncodeMediaTypeOverride.java[]
```

Comment:
@gustavonalle I think this [code example](https://github.com/infinispan/infinispan/blob/master/documentation/src/main/asciidoc/topics/code_examples/EncodeMediaTypeOverride.java) needs to be updated.

please correct my understanding but could we update the example to show application/x-protostream instead of application/x-java-object in the ConfigurationBuilder? then show application/x-java-object as the override?

Reply:
This section was not part of the protocol interoperability? I need to check the context where it is inserted then I will let you know

Reply:
thanks @gustavonalle tbh I find this topic is a little confusing. what I've been trying to do in this PR is separate the details about mediatype encoding into distinct procedures. 

in the current docs, mediatype configuration is included in the section about encoders/transcoders. IMO those details are a little advanced and can make it difficult to understand for users who are trying to get up and running. so I think it makes sense to move the procedure for configuring mediatypes to the protocol interoperability section. maybe we should move this back to the encoding section as well if it's for more advanced cases.